x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/json_diff/diff_view.tsx

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+import { css, Global } from '@emotion/react';
+import {
+  Diff,
+  useSourceExpansion,
+  useMinCollapsedLines,
+  parseDiff,
+  tokenize,
+} from 'react-diff-view';
+import 'react-diff-view/style/index.css';
+import type { RenderGutter, HunkData, TokenizeOptions, DiffProps } from 'react-diff-view';
+import unidiff from 'unidiff';
+import { useEuiTheme } from '@elastic/eui';
+import { Hunks } from './hunks';
+import { markEdits, DiffMethod } from './mark_edits';
+
+const useExpand = (hunks: HunkData[], oldSource: string, newSource: string) => {
+  const [hunksWithSourceExpanded, expandRange] = useSourceExpansion(hunks, oldSource);
+  const hunksWithMinLinesCollapsed = useMinCollapsedLines(0, hunksWithSourceExpanded, oldSource);
+
+  return {
+    expandRange,
+    hunks: hunksWithMinLinesCollapsed,
+  };
+};
+
+const useTokens = (hunks: HunkData[], diffMethod: DiffMethod, oldSource: string) => {
+  if (!hunks) {
+    return undefined;
+  }
+
+  const options: TokenizeOptions = {
+    oldSource,
+    highlight: false,
+    enhancers: [
+      /*
+        "markEditsBy" is a slightly modified version of "markEdits" enhancer from react-diff-view
+        to enable word-level highlighting.
+      */
+      markEdits(hunks, diffMethod),
+    ],
+  };
+
+  try {
+    /*
+      Synchroniously applies all the enhancers to the hunks and returns an array of tokens.
+      There's also a way to use a web worker to tokenize in a separate thread.
+      Example can be found here: https://github.com/otakustay/react-diff-view/blob/49cebd0958ef323c830395c1a1da601560a71781/site/components/DiffView/index.tsx#L43
+      It didn't work for me right away, but theoretically the possibility is there.
+    */
+    return tokenize(hunks, options);
+  } catch (ex) {
+    return undefined;
+  }
+};
+
+const renderGutter: RenderGutter = ({ change }) => {
+  /*
+    Custom gutter (a column where you normally see line numbers).
+    Here's I am returning "+" or "-" so the diff is more readable by colorblind people.
+  */
+  if (change.type === 'insert') {
+    return <span>{'+'}</span>;
+  }
+
+  if (change.type === 'delete') {
+    return <span>{'-'}</span>;
+  }
+
+  if (change.type === 'normal') {
+    return null;
+  }
```

Comment:
nit: `return null` should suffice right?

Or can `change.type` take any other values apart from these three?